### ISSUE TITLE
Nothing displayed, but Right-Click works. Some computers only.

### ISSUE THREAD

**Issue by TcNobo** [MicrosoftEdge/WebView2Feedback#1539]
This project of mine, the [TcNo Account Switcher](https://github.com/TcNobo/TcNo-Acc-Switcher) is being plagued by an issue that affects... some people.
The WebView2 window works perfectly. A handful of computers display nothing but the form's background, even when the page is changed, and can be interacted with (Both elements on the page, and the Right-Click window for the Inspector).

It affects both [Windows 7 - issue #45](https://github.com/TcNobo/TcNo-Acc-Switcher/issues/45#issuecomment-874480437) and Windows 10 users. Windows 10 I can't link as they contacted via Discord, but it's been 2 or 3 users so far, out of a total of around 25,000.
Nothing is different in the builds. They work, and on some computers, they don't. I can't recreate it, but I need to solve it.

I tried 'fixes' from https://github.com/MicrosoftEdge/WebView2Feedback/issues/1077, in https://github.com/TcNobo/TcNo-Acc-Switcher/commit/1c085cede0efa315202a1a985271fd8d179eb696. But they came to the same result. Not fixing anything.

**Version**
SDK: WebView2 1.0.864.35, WebView 2 1.0.902-prerelease. ASP.NET Core + Desktop Runtime 5.0.8.
Runtime: WebView2 Runtime 91.0.864.70 (Latest)
Framework: WinForms
OS: Windows 10 latest, Windows 7 latest.

**Repro Steps**
I have no steps as nothing is done differently on computers where it works, and where it doesn't. Some just don't display anything, some do. It should open up and display everything as normal, but for some, it just doesn't.

**Screenshots**
Expected: 
![image](https://user-images.githubusercontent.com/10319195/126039095-a21486b7-790f-41af-9a74-26bcff8b79f2.png)
Actual, on PCs affected by this bug: 
![image](https://user-images.githubusercontent.com/10319195/126039099-bffbbc7f-6459-4db7-a361-b38d28e10a62.png)

**Additional context**
You're able to download it yourself, [here](https://github.com/TcNobo/TcNo-Acc-Switcher/releases/tag/2021-07-14_00). The installer installs all required runtimes, so nobody is missing anything causing this issue. Sometimes users missing 1 of the 3 required runtimes get the issue of nothing displaying, but everyone with this issue has all runtimes installed.

**Comment by champnic** [comment]
Hey @TcNobo - thanks for the bug report! A couple of questions:
1. Are you setting the WebView2 to have a transparent background?
2. Is this happening when minimizing and then opening the app?
3. Are you able to find out what GPUs the affected machines are using?

We are currently investigating similar bugs, so I'll link your issue to those investigations. Thanks!

**Comment by TcNobo** [comment]
1. No.
2. Just when opening the program, and lasts throughout its life.
3. The 2 computers affected:
- Windows 10 Education x64 19042, Polish, Intel(R) HD Graphics 4000. See [Imgur album](https://imgur.com/a/7Wu21cZ)
- Windows 7 (Maximum?) x64 7601, Russian, Intel HD Graphics. See https://github.com/TcNobo/TcNo-Acc-Switcher/issues/45#issuecomment-884352236

I asked users for DXDiag screenshots, so that's what the links are for if you can pull any more useful info out of it.

**Comment by champnic** [comment]
Thanks for the info! I'm adding this bug to our backlog. We have some similar bugs on our backlog that we're working on and are related to DComp being disabled for certain GPUs, which seems to be the case for the Intel 4000 and OS < Win10.

**Comment by TcNobo** [comment]
Thanks. Are there any known workarounds that I can use in the meanwhile?

**Comment by champnic** [comment]
It does look like in those cases that DComp would be disabled, and so it probably is the same issue we are already investigating. We are working on fixes for that now. Thanks!

**Comment by champnic** [comment]
Unfortunately we haven't found workaround yet for folks who find themselves in this state.

**Comment by TcNobo** [comment]
Have another user with the same exact issue. Same thing, also Intel HD Graphics:
- Windows 7 Ultimate x64 7601, Turkish, Intel i5-4300M (Intel HD Graphics 4600). See [Imgur album](https://imgur.com/a/0GclDma)

**Comment by TcNobo** [comment]
For now I've worked around this by building in a 'fallback' CEFSharp version. Extra files are downloaded, and the program uses this instead of WebView2.
I much prefer the performance of WebView2 - and the Inspect Element functionality... I do hope this gets patched soon.

**Comment by markismail7** [comment]
I'm having the same problem with some of our customers. Webview2 won't work, i even had the customer upgrade from windows 10 to windows 11 and still didn't fix it. It's 64 bit computer. I hope Microsoft takes care of it as we have no answer for our customers on why certain computers can't load webview2

**Comment by champnic** [comment]
@markismail7 These issues are tough to repro and track down, but in my experience they are most often tied to GPU driver issues. If you want to, you can try disabling the GPU workarounds that exist for some driver bugs. This is a tradeoff in that you may hit other graphical issues tied to drivers, but may avoid the content being invisible issue. To try this, add these to the browser arguments when initializing the WebView2:
`--ignore-gpu-blocklist --disable-gpu-driver-bug-workarounds`

**Comment by markismail7** [comment]
@champnic How do you answer everyone's question so fast. OMG. Microsoft give this person a raise. Thank you i'll try that. I really appreciate your help.

**Comment by markismail7** [comment]
@champnic 
I haven't been able to solve the problem on certain computers where webview2 doesn't load. Its just white blank. However, they can load edge chrome. Can you please help, i have encountered over 50 customers for the past 30 days. Some of them have 2 computers, one can load webview2 the other can't load it.

**Comment by champnic** [comment]
Are you suddenly seeing a surge of this occurring?

For the machines that are hitting this, do the following browser arguments "fix" the issue and allow WebView2 to load? You can set the following environment variable of those machines and restart the WebView2/app:
`WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS=--ignore-gpu-blocklist --disable-gpu-driver-bug-workarounds`

**Comment by markismail7** [comment]
Yes, they told me it was working before

**Comment by markismail7** [comment]
I'm working with a customer right now, i'm trying to find the error. I'll post it soon

**Comment by markismail7** [comment]
Here is the error that comes up when initializing WEbview2:
"Unable to cast to Mircrosoft.web.webview2.core.raw.icorewebview2controller2. This may happen if you are using an interface not supported by the version of the webview2 runtime you are using. "

**Comment by markismail7** [comment]
I'm using the lastest version of webview2 production, it seems like the for some computer the webview2 runtime is not matching

**Comment by markismail7** [comment]
@champnic 
So i found the problem, it exactly what it says. The webview2 runtime is outdated and not matching webview2 core. I'm downloading it from this url https://developer.microsoft.com/en-us/microsoft-edge/webview2/  (Evergreen standalone)


I had to downgrade webview2 from :  1.0.1185.39     to  1.0.1020.30 . Please let me if i'm suppose to download runtime from somewhere else or its runtime on this link needs to be updated it. Thanks Mark

**Comment by champnic** [comment]
Thanks for digging in @markismail7 - sounds like this is a different issue than the main thread here (which has nothing getting displayed, but the WebView2 is loading fine).

ICoreWebView2Controller2 is an old interface (1.5 yrs) supported by runtimes 89.0.774.45+. What runtime did the user have on their machine? In theory the evergreen runtime should be keeping itself up to date. Did downgrading the SDK but keeping the same old runtime workaround the issue?

**Comment by markismail7** [comment]
@champnic 
No problem, so i tried to upgrade the runtime, it kept saying, it's already installed. Here is the issue, this only happens with certain computers. The runtime is not working with certain computers with the lates version of webview2. Once i downgraded webview2, it loaded perfectly. Other customer didn't encounter this issue.

**Comment by champnic** [comment]
Can you get version info of the SDK and Runtimes for machines where you are and aren't seeing this?

**Comment by markismail7** [comment]
I will get it.

**Comment by markismail7** [comment]
@champnic 
It is the latest version: 101.0.1210.32

**Comment by champnic** [comment]
Just to confirm - that is the version combination that you are seeing it fail due to the ICoreWebView2Controller2?
"Unable to cast to Mircrosoft.web.webview2.core.raw.icorewebview2controller2."

**Comment by markismail7** [comment]
Yes, keep in mind. This only happens on certain computers. So it doesn't make any sense. Another thing i found out, if you have webview2 runtime older version installed, and you try to install the new version from https://developer.microsoft.com/en-us/microsoft-edge/webview2/  It will say, you already have webview2 installed. It doesn't force to update. You need to un install and re- install to have the latest version. I know it suppose to update automatically though.
